### PR TITLE
feat: Migrate Redux to TS

### PR DESCRIPTION
**Pull request by przemvs**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: electron/renderer/src/components/WebView/Webview.tsx [hunk 1]
```diff
@@ -37,22 +37,34 @@ import {
...
+
+export const isNumber = (value: any): value is number => typeof value === 'number';
+
+export const isBoolean = (value: any): value is boolean => typeof value === 'boolean';
+
+export const isAccount = (value: any): value is Account => value.hasOwnProperty('userID');
+
+export const isConversationJoinData = (value: any): value is ConversationJoinData =>
+  value.hasOwnProperty('code') && value.hasOwnProperty('key');
+
```
Comment (PatrykBuniX) [inline electron/renderer/src/components/WebView/Webview.tsx:60]: I think these should live in a separate file

File: tsconfig.json [hunk 2]
```diff
@@ -12,22 +12,27 @@
...
+      "src/*": ["electron/src/*", "electron/renderer/*"]
+    },
     "skipLibCheck": true,
     "sourceMap": true,
     "strict": true,
     "target": "es6",
     "typeRoots": ["electron/src/types", "node_modules/@types"],
     "jsx": "react-jsx"
   },
-  "exclude": ["bin", "electron/dist", "electron/**/*.test*.ts", "node_modules", "wrap"],
```
Comment (PatrykBuniX) [inline tsconfig.json:30]: Should `"bin"` be included now?

File: package.json [hunk 3]
```diff
@@ -181,7 +182,7 @@
     "lint:other": "yarn prettier --list-different",
     "lint:spelling": "cspell -c .cspell.json '*.md' 'electron/renderer/**/*.jsx' 'electron/src/**/*.ts' 'electron/src/**/*.tsx' 'electron/html/*.html'",
     "postinstall": "yarn configure",
-    "prestart": "yarn build:ts && yarn bundle:dev",
+    "prestart": "yarn build:ts && yarn bundle",
```
Comment (PatrykBuniX) [inline package.json:185]: Is this intentional?

File: electron/renderer/src/actions/AccountAction.ts [hunk 4]
```diff
@@ -19,42 +19,58 @@
...
+      getState: () => State,
+      {actions: {accountAction}}: {actions: {accountAction: AccountActionTypes}},
+    ) => {
       try {
         const accounts = AccountSelector.getAccounts(getState());
         const loggedOutWebviews = accounts.filter(account => account.userID === undefined);
 
         if (loggedOutWebviews.length > 0) {
+          // @ts-ignore
           dispatch(accountAction.switchWebview(accounts.indexOf(loggedOutWebviews[0])));
```
Comment (PatrykBuniX) [inline electron/renderer/src/actions/AccountAction.ts:59]: Can figure sth out to get rid of all these `@ts-ignore`s?

File: electron/renderer/src/actions/AccountAction.ts [hunk 5]
```diff
@@ -19,42 +19,58 @@
...
 
           const accountId = loggedOutWebviews[0].id;
           dispatch(initiateSSO(accountId, ssoCode, accounts.length == 1));
         } else {
           if (accounts.length >= config.maximumAccounts) {
-            return window.dispatchEvent(
+            window.dispatchEvent(
```
Comment (PatrykBuniX) [inline electron/renderer/src/actions/AccountAction.ts:65]: we can do `return void window.dispatchEvent(...`

File: electron/renderer/src/selector/AccountSelector.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
+/*
+ * Wire
+ * Copyright (C) 2020 Wire Swiss GmbH
```
Comment (thisisamir98) [inline electron/renderer/src/selector/AccountSelector.ts:3]:
```suggestion
 * Copyright (C) 2023 Wire Swiss GmbH
```

File: electron/renderer/src/selector/ContextMenuSelector.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Wire
+ * Copyright (C) 2020 Wire Swiss GmbH
```
Comment (thisisamir98) [inline electron/renderer/src/selector/ContextMenuSelector.ts:3]:
```suggestion
 * Copyright (C) 2023 Wire Swiss GmbH
```

File: electron/renderer/src/selector/AccountSelector.ts (hunk 6 above)
Comment (V-Gira) [reply electron/renderer/src/selector/AccountSelector.ts:3]: dammit beat me on the only cr I can do :D

**Review by thisisamir98 — APPROVED** [review]

File: electron/wire.json [hunk 8]
```diff
@@ -15,8 +15,8 @@
   "legalUrl": "https://wire.com/legal/",
   "licensesUrl": "https://wire.com/legal/licenses/",
   "maximumAccounts": "3",
-  "name": "WireInternal",
-  "nameShort": "WireInternal",
+  "name": "Wire",
+  "nameShort": "Wire",
```
Comment (tlebon) [inline electron/wire.json:19]: can we leave this as it was previously?

**Review by tlebon — APPROVED** [review]

**Review by PatrykBuniX — APPROVED** [review]

**Pull request merged**